### PR TITLE
Show syntax lexer name in file view/blame

### PR DESCRIPTION
**Pull request by silverwind**
Show which Chroma Lexer is used to highlight the file in the file header. It's useful for development to see what was detected, and I think it's not bad info to have for the user:

Screenshot 2022-11-14 at 22 31 16
Screenshot 2022-11-14 at 22 36 06
Screenshot 2022-11-14 at 22 36 26

Also, I improved the way this header overflows on small screens:

Screenshot 2022-11-14 at 22 44 36

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Of course you know what the next issue is going to be... 

I suppose it should be possible to pass a requested lexer name in as request parameter.

**Comment by silverwind** [comment]
Small tweak, Chroma has a `fallback` lexer, that uses plaintext. Map it to that name:

https://github.com/alecthomas/chroma/blob/f2bd8424a3ccbf7a8c574855f060049826c7decc/lexers/lexers.go#L69-L73

File: templates/repo/blame.tmpl [hunk 1]
```diff
@@ -1,14 +1,9 @@
...
-		<div class="file-header-left df ac">
-			<div class="file-info text grey normal mono">
-				<div class="file-info-entry">
-					{{.NumLines}} {{.locale.TrN .NumLines "repo.line" "repo.lines"}}
-				</div>
-				<div class="file-info-entry">{{FileSize .FileSize}}</div>
-			</div>
+	<h4 class="file-header ui top attached header df ac sb fw">
+		<div class="file-header-left df ac py-3 pr-4">
+			{{template "repo/file_info" .}}
```
Comment (delvh) [inline templates/repo/blame.tmpl:4]: Was the template here previously outdated, or were the other entries missing on purpose?

File: templates/repo/file_info.tmpl [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<div class="file-info text grey normal mono">
+	{{if .FileIsSymlink}}
+		<div class="file-info-entry">
+			{{.locale.Tr "repo.symbolic_link"}}
+		</div>
+	{{end}}
+	{{if .NumLinesSet}}
```
Comment (delvh) [inline templates/repo/file_info.tmpl]: Isn't that attribute a bit overkill, if we could use
```suggestion
	{{if .NumLines}}
```
just as well?

File: modules/highlight/highlight_test.go [hunk 3]
```diff
@@ -78,10 +78,11 @@ c=2
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			out, err := File(tt.name, "", []byte(tt.code))
+			out, lexerName, err := File(tt.name, "", []byte(tt.code))
 			assert.NoError(t, err)
 			expected := strings.Join(tt.want, "\n")
 			actual := strings.Join(out, "\n")
+			assert.NotEmpty(t, lexerName)
```
Comment (delvh) [inline modules/highlight/highlight_test.go]: Should we perhaps enhance the test to test if the expected lexer has been found?

File: templates/repo/blame.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/blame.tmpl:4]: I think it was outdated and people forgot to keep blame fields in sync with view fields. At least, I think the checks for the new fields shouldn't cause any issues if absent.

File: templates/repo/file_info.tmpl (hunk 2 above)
Comment (silverwind) [reply templates/repo/file_info.tmpl]: `.NumLinesSet` is necessary so we can show 0 `.NumLines`.

File: modules/highlight/highlight_test.go (hunk 3 above)
Comment (silverwind) [reply modules/highlight/highlight_test.go]: Hmm yes, we could add that.

File: templates/repo/file_info.tmpl (hunk 2 above)
Comment (delvh) [reply templates/repo/file_info.tmpl]:
```suggestion
	{{if .NumLinesSet}}{{/* Explicit attribute needed to show 0 line changes */}}
```

File: modules/highlight/highlight_test.go (hunk 3 above)
Comment (silverwind) [reply modules/highlight/highlight_test.go]: Done

**Review by delvh — APPROVED** [review]

**Review by yardenshoham — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Fixed formatting. One noteworthy additional change is the introduction of `ToTitleCaseNoLower` in https://github.com/go-gitea/gitea/pull/21814/commits/66ef6e226b8a59891171f3b9df95881b8ce8aacb. It adds the `cases.NoLower` option so `JavaScript` does not become `Javascript`.

**Comment by lafriks** [comment]
🚀

**Pull request merged**